### PR TITLE
Support Spring

### PR DESCRIPTION
**Pull request by bauersimon**

### CODE REVIEW

File: task/config.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package task
```
Comment (ruiAzevedo19) [inline task/config.go:1]: Putting this in `task/config.go`  sounds like this is a task configuration but it is rather a repository configuration right? Should the file be renamed from `config.go` to `repository.go`? Or even put this in a `repository` package e.g. `repository/config.go`?

File: model/symflower/symflower.go [hunk 2]
```diff
@@ -16,34 +16,55 @@ import (
...
+}
+
+// NewModelSymbolicExecution returns a symbolic execution Symflower model.
+func NewModelSymbolicExecution() (model *Model) {
+	return NewModelSymbolicExecutionWithTimeout(defaultSymflowerTimeout)
 }
 
-// NewModel returns a Symflower model.
-func NewModel() (model *Model) {
+// NewModelSymbolicExecutionWithTimeout returns a Symflower model with a given timeout.
```
Comment (ruiAzevedo19) [inline model/symflower/symflower.go:38]: `returns a symbolic execution Symflower model`

File: model/symflower/symflower.go [hunk 3]
```diff
@@ -16,34 +16,55 @@ import (
...
+		symbolicExecutionTimeout: timeout,
+
+		command: "unit-tests",
+		id:      "symbolic-execution",
 	}
 }
 
-// NewModelWithTimeout returns a Symflower model with a given timeout.
-func NewModelWithTimeout(timeout time.Duration) (model *Model) {
+// NewModelSmartTemplate returns a symbolic execution Symflower model.
```
Comment (ruiAzevedo19) [inline model/symflower/symflower.go:48]: This is not a symbolic execution model but rather test templates.

File: model/symflower/symflower_test.go [hunk 4]
```diff
@@ -139,6 +144,27 @@ func TestModelGenerateTestsForFile(t *testing.T) {
 			},
 		})
 	})
+	validate(t, &testCase{
+		Name: "Skeleton",
```
Comment (ruiAzevedo19) [inline model/symflower/symflower_test.go:148]: Maybe `Smart template`?

File: README.md [hunk 5]
```diff
@@ -226,6 +226,17 @@ Each repository can contain a configuration file `repository.json` in its root d
...
+Depending on the task, it can be beneficial to exclude parts of the repository from explicit evaluation. To give a concrete example: Spring controller tests can never be executed on their own but need a supporting [`Application` class](https://docs.spring.io/spring-boot/reference/testing/spring-boot-applications.html#testing.spring-boot-applications.using-main). But such a file should never be used itself to prompt models for tests. Therefore, it can be excluded through the `repository.json` configuration:
+
+```json
+{
+  "tasks": ["write-tests"],
+  "ignore": ["src/main/java/com/example/Application.java"]
+}
+```
+
+This `ignore` setting is currently only respected for the test generation task `write-tests`.
```
Comment (ruiAzevedo19) [inline README.md:238]: `... is currently applicable only for the...` maybe?

File: task/config.go (hunk 1 above)
Comment (bauersimon) [reply task/config.go:1]: The interface for `Repository` is also defined within `task`. That's why I moved it there. So we would probably want to move that one as well. Would leave it for now.

Comment (ruiAzevedo19) [reply task/config.go:1]: Let's leave it for now then

**Review by ruiAzevedo19 — APPROVED** [review]

**Pull request merged**